### ISSUE TITLE
Why is Dir.children documented for Ruby 2.4.1?

### ISSUE THREAD

**Issue by meineerde** [documenting-ruby/ruby#81]
The `Dir.children` method was added to Ruby trunk with [Feature #11302](https://bugs.ruby-lang.org/issues/11302). It was however not backported to Ruby 2.4 yet. Attempting to use it results in a `NoMethodError` in Ruby.

However, the method documentation is added to the Ruby 2.4.1 documentation at https://ruby-doc.org/core-2.4.1/Dir.html#method-c-children. Why is this the case? The method is not part of Ruby 2.4 and is (as far as I can see) not mentioned in the source code.

This was also asked and investigated at https://stackoverflow.com/a/45719685/421705

**Comment by chrisarcand** [comment]
Indeed, `Dir.each_child` as well, and it's present in all 2.4.x documentation. Added [here](https://bugs.ruby-lang.org/projects/ruby-trunk/repository/revisions/58879/diff/dir.c).

**Comment by meineerde** [comment]
Similarly, `Hash#transform_keys` was added to trunk in [Feature #13583](https://bugs.ruby-lang.org/issues/13583) but not (yet) to Ruby 2.4. See https://stackoverflow.com/a/46414193/421705

**Comment by eregon** [comment]
We noticed a similar issue in https://github.com/ruby/spec/pull/483.
It seems the docs ruby for 2.4.0, 2.4.1 and 2.4.2 all use trunk to generate the html documentaion, while they should of course use the corresponding tags like v2_4_2.

Same issue as #80.
@zzak do you know who we should ping about this or where to open an issue for it?
